File: mns-core/src/test/java/cz/fb/manaus/core/model/ModelFactory.java
package cz.fb.manaus.core.model;

import java.util.Collection;
import java.util.Date;

public interface ModelFactory {
    static BetAction newAction(BetActionType betActionType, Date actionDate, Price price,
                               Market market, long selectionId) {
        var ba = new BetAction();
        ba.setBetActionType(betActionType);
        ba.setActionDate(actionDate);
        ba.setPrice(price);
        ba.setMarket(market);
        ba.setSelectionId(selectionId);
        return ba;
    }

    static Event newEvent(String id, String name, Date openDate, String countryCode) {
        var event = new Event();
        event.setId(id);
        event.setName(name);
        event.setOpenDate(openDate);
        event.setCountryCode(countryCode);
        return event;
    }

    static MarketPrices newPrices(int winnerCount, Market market, Collection<RunnerPrices> runnerPrices, Date time) {
        var mp = new MarketPrices();
        mp.setWinnerCount(winnerCount);
        mp.setMarket(market);
        mp.setRunnerPrices(runnerPrices);
        mp.setTime(time);
        return mp;
    }

    static Runner newRunner(long selectionId, String name, double handicap, int sortPriority) {
        var runner = new Runner();
        runner.setSelectionId(selectionId);
        runner.setName(name);
        runner.setHandicap(handicap);
        runner.setSortPriority(sortPriority);
        return runner;
    }

    static RunnerPrices newRunnerPrices(long selectionId, Collection<Price> prices, Double matched, Double lastMatchedPrice) {
        var rp = new RunnerPrices();
        rp.setSelectionId(selectionId);
        rp.setPrices(prices);
        rp.setMatchedAmount(matched);
        rp.setLastMatchedPrice(lastMatchedPrice);
        return rp;
    }

    static SettledBet newSettled(long selectionId, String selectionName, double profitAndLoss, Date settled, Price price) {
        var bet = new SettledBet();
        bet.setSelectionId(selectionId);
        bet.setSelectionName(selectionName);
        bet.setProfitAndLoss(profitAndLoss);
        bet.setSettled(settled);
        bet.setPrice(price);
        return bet;
    }
}

File: mns-core/src/test/java/cz/fb/manaus/core/test/ModelFactory.kt
package cz.fb.manaus.core.test

import cz.fb.manaus.core.model.*
import java.util.*

interface ModelFactory {
    companion object {
        fun newAction(betActionType: BetActionType, actionDate: Date, price: Price,
                      market: Market, selectionId: Long): BetAction {
            val ba = BetAction()
            ba.betActionType = betActionType
            ba.actionDate = actionDate
            ba.price = price
            ba.market = market
            ba.selectionId = selectionId
            return ba
        }

        fun newEvent(id: String, name: String, openDate: Date, countryCode: String): Event {
            val event = Event()
            event.id = id
            event.name = name
            event.openDate = openDate
            event.countryCode = countryCode
            return event
        }

        fun newPrices(winnerCount: Int, market: Market, runnerPrices: Collection<RunnerPrices>, time: Date): MarketPrices {
            val mp = MarketPrices()
            mp.winnerCount = winnerCount
            mp.market = market
            mp.runnerPrices = runnerPrices
            mp.time = time
            return mp
        }

        fun newRunner(selectionId: Long, name: String, handicap: Double, sortPriority: Int): Runner {
            val runner = Runner()
            runner.selectionId = selectionId
            runner.name = name
            runner.handicap = handicap
            runner.sortPriority = sortPriority
            return runner
        }

        fun newRunnerPrices(selectionId: Long, prices: Collection<Price>, matched: Double?, lastMatchedPrice: Double?): RunnerPrices {
            val rp = RunnerPrices()
            rp.selectionId = selectionId
            rp.prices = prices
            rp.matchedAmount = matched
            rp.lastMatchedPrice = lastMatchedPrice
            return rp
        }

        fun newSettled(selectionId: Long, selectionName: String, profitAndLoss: Double, settled: Date, price: Price): SettledBet {
            val bet = SettledBet()
            bet.selectionId = selectionId
            bet.selectionName = selectionName
            bet.profitAndLoss = profitAndLoss
            bet.settled = settled
            bet.price = price
            return bet
        }
    }
}
